File: 111.MinimumDepthOfBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if root == None:
            return 0
        
        ld = self.minDepth(root.left)
        rd = self.minDepth(root.right)

        if ld == 0:
            return rd + 1

        if rd == 0:
            return ld + 1

        return min(ld, rd) + 1

File: 111.MinimumDepthOfBinaryTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {

        if (root == null) {
            return 0;
        }

        int ld = minDepth(root.left);
        int rd = minDepth(root.right);

        if (ld == 0) {
            return rd + 1;
        }

        if (rd == 0) {
            return ld + 1;
        }

        return Math.min(ld, rd) + 1;
        
    }
}
